src/joinGame.js

```diff
@@ -20,8 +20,76 @@ const checkGameExists = (gameId) => {
     return idExists;
 };
 
-const checkGameNotFull = (gameId) => {};
+const checkGameNotFull = (interaction, gameId) => {
+    const game = gamesIndex[gameId];
+    let gameFull = false;
 
-export const joinGame = (interaction, gameId) => {
-    if (!checkGameExists(gameId) || !checkGameNotFull(gameId)) return;
+    if (game.guesser && game.giver) {
+        gameFull = true;
+        interaction.reply({
+            ephemeral: true,
+            content: `Game is already full! Please try to join another game.`,
+        });
+    }
+
+    return gameFull;
+};
+
+const checkGameUsers = (interaction, gameId) => {
+    const game = gamesIndex[gameId];
+    const user = interaction.user.username;
+    let usersUnique = true;
+
+    if (user === game.guesser || user === game.giver) {
+        usersUnique = false;
+        interaction.reply({
+            ephemeral: true,
+            content: `You can't play against yourself!`,
+        });
+    }
+
+    return usersUnique;
+};
+
+export const joinGame = async (interaction, gameId) => {
+    const game = gamesIndex[gameId.toString()];
+
+    // Swap to this if check once testing complete
+    //   if (
+    //     !checkGameExists(interaction, gameId) ||
+    //     checkGameNotFull(interaction, gameId) ||
+    //     checkGameUsers(interaction, gameId)
+    // )
+
+    if (
+        !checkGameExists(interaction, gameId) ||
+        checkGameNotFull(interaction, gameId)
+    )
+        return;
+
+    if (!game.giver) {
+        game.giver = interaction.user.username;
+        game.giverId = interaction.user.id;
+    } else {
+        game.guesser = interaction.user.username;
+        game.guesserId = interaction.user.id;
+    }
+
+    gamesIndex.gameState = "playing";
```

Comment:
this "playing" state, just for consistency sake, should be part of the `GAME_STATE` object in `gameIndex.js`

Reply:
good point. i've amended `PENDING` to `STATUS` to avoid having something like:

`export const GAME_STATE = {
    PENDING: "playing",
};`

Reply:
sorry I should have been a bit clearer! with the suggested change it was more along the lines of `GAME_STATE` being:
```js
export const GAME_STATE = {
    PENDING: "pending",
    PLAYING: "playing",
}
```
and then that line being:
`gamesIndex.gameState = GAME_STATE.PLAYING;`
because we still want the status of games to be per-game